File: LeetCode/Easy/1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        end = len(grid[0])
        ret = 0
        
        for row in grid:
            i, j = 0, end - 1
            while i <= j:
                mid = (i + j) // 2
                if row[mid] > -1:
                    i = mid + 1
                else:
                    j = mid - 1

            ret += (end - i)

        return ret

File: LeetCode/Easy/1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int end = grid[0].length;
        int ret = 0;
        int left;
        int right;
        int mid;
        for (int row = 0; row < grid.length; ++row){
            left = 0;
            right = end - 1;
            while (left <= right){
                mid = (left + right) / 2;
                
                if (grid[row][mid] > -1) left = mid + 1;
                else right = mid - 1;
            }
            
            ret += end - left;
        }
        
        return ret;
    }
}